app/jobs/notify_job.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class NotifyJob < ApplicationJob
+  queue_as :webhooks
+
+  def perform(notification_id:)
+    notification = Notification.kept.find(notification_id)
+    raise 'Subscription disabled' unless notification.subscription.enabled?
+
+    data = ActiveModelSerializers::Adapter.create(NotificationSerializer.new(notification)).to_json
+    hmac = Encryptor.hmac(notification.subscription.secret, data)
+
+    response = client.post(notification.subscription.callback_url, data, 'PECS-SIGNATURE': hmac, 'PECS-NOTIFICATION-ID': notification_id)
+
+    notification.update(delivered_at: DateTime.now) if response.success?
```

Comment:
Yes we need it: delivered_at is basically saying the notification has been delivered successfully, under this line it updates the timestamps ALSO in case the notification wasn't a success.

Reply:
ah ok, I got it.
Could this work, so you can use just an update? 
```
notification.delivered_at = DateTime.now if response.success?
assign_attributes(delivery_attempted_at: DateTime.now, delivery_attempts: notification.delivery_attempts.succ)
notification.save
```